File: workhive-backend/internal/models/message.go
package models

import "github.com/google/uuid"

type Message struct {
	Base
	ContractID uuid.UUID `gorm:"type:uuid;not null;index" json:"contract_id"`
	SenderID   uuid.UUID `gorm:"type:uuid;not null;index" json:"sender_id"`
	Content    string    `gorm:"type:text;not null" json:"content"`
	IsRead     bool      `gorm:"default:false" json:"is_read"`

	// associations
	Contract Contract `gorm:"foreignKey:ContractID" json:"contract,omitempty"`
	Sender   User     `gorm:"foreignKey:SenderID" json:"sender,omitempty"`
}

File: workhive-frontend/src/types/message.ts
import type { UserBrief } from "./auth"

export interface Message {
  id: string
  contract_id: string
  sender_id: string
  content: string
  is_read: boolean
  created_at: string
  sender?: UserBrief
}

export interface SendMessageRequest {
  content: string
}
